com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -8,101 +8,360 @@
+                {
+                    return;
+                }
+                SceneBeingSynchronized = SceneManager.GetActiveScene();
+                return;
+            }
+
+            var clientSceneHandle = serverSceneHandle;
+            if (m_NetworkManager.SceneManager.ServerSceneHandleToClientSceneHandle.ContainsKey(serverSceneHandle))
+            {
+                clientSceneHandle = m_NetworkManager.SceneManager.ServerSceneHandleToClientSceneHandle[serverSceneHandle];
+                // If we were already set, then ignore
+                if (SceneBeingSynchronized.IsValid() && SceneBeingSynchronized.isLoaded && SceneBeingSynchronized.handle == clientSceneHandle)
+                {
+                    return;
+                }
+
+                // Find and set the scene currently being synchronized
+                SceneBeingSynchronized = new Scene();
+                foreach (var keyValuePairBySceneName in m_ScenesLoaded)
+                {
+                    if (keyValuePairBySceneName.Value.ContainsKey(clientSceneHandle))
+                    {
+                        SceneBeingSynchronized = keyValuePairBySceneName.Value[clientSceneHandle];
+                    }
+                }
+
+                if (!SceneBeingSynchronized.IsValid() || !SceneBeingSynchronized.isLoaded)
+                {
+                    throw new Exception($"[{nameof(NetworkSceneManager)}- {nameof(m_ScenesLoaded)}] Could not find the appropriate scene to set as being synchronized!");
+                }
+            }
+            else
+            {
+                // This should never happen, but in the event it does...
+                throw new Exception($"[{nameof(SceneEventData)}- Scene Handle Mismatch] {nameof(serverSceneHandle)} could not be found in {nameof(ServerSceneHandleToClientSceneHandle)}!");
+            }
+        }
+
+        /// <summary>
+        /// During soft synchronization of in-scene placed NetworkObjects, this is now used by NetworkSpawnManager.CreateLocalNetworkObject
+        /// </summary>
+        /// <param name="globalObjectIdHash"></param>
+        /// <returns></returns>
+        internal NetworkObject GetSceneRelativeInSceneNetworkObject(uint globalObjectIdHash)
+        {
+            if (ScenePlacedObjects.ContainsKey(globalObjectIdHash))
+            {
+                if (ScenePlacedObjects[globalObjectIdHash].ContainsKey(SceneBeingSynchronized.handle))
                 {
-                    NetworkLog.LogWarning($"The current scene ({SceneManager.GetActiveScene().name}) is not regisered as a network scene.");
+                    var inScenePlacedNetworkObject = ScenePlacedObjects[globalObjectIdHash][SceneBeingSynchronized.handle];
+
+                    // We can only have 1 duplicated globalObjectIdHash per scene instance, so remove it once it has been returned
+                    ScenePlacedObjects[globalObjectIdHash].Remove(SceneBeingSynchronized.handle);
```

Comment:
This feels like something that should be done by the caller as it seems specific to how it is called, rather than a generic "get network object relative to the scene" function that the name might imply.  Could we separate this into its own function that NetworkSpawnManager.CreateLocalNetworkObject could use?

Reply:
This actually is a very unique scenario that is synchronizing in-scene placed NetworkObjects.  This falls into the NetworkSceneManager realm because the NetworkObjects in question are instantiated by the loading of a level and not by the NetworkSpanwManager.  NetworkSceneManager.GetSceneRelativeInSceneNetworkObject replaces this section of code in NetworkSpawnManager that was formerly directly accessing a list of the in scene placed network objects that was within NetworkSceneManager.  Since the end result was to get the in-scene placed NetworkObject in question and due to the fact that we can now have the same additive scene loaded multiple times, it seemed easier to just make it a function call in NetworkSceneManager that handles figuring out which instance of the in-scene place NetworkObject (additive scene instance relative) to return.

Did this help better explain why it is the way that it is?

Reply:
So is this removing the SceneBeingSynchronized scene from the ScenePlacedObjects listas it's now been loaded into the correct instance of the scene (and not the temporary scene?)

Reply:
Yes let's chat, and don't let this comment stop the review going through, I don't think it's anything serious.

I understand the need to distinguish network objects by their scene handle due to additive scene loading meaning you might have the same scene twice, the bit that's tripping me up is why we're removing the scene handle here, I think I get it but I want to confirm what I'm thinking!